content/guides/references/migration-guide.md

```diff
@@ -2,6 +2,345 @@
+to write independent, reliable and deterministic tests from the start.
+
+If for whatever reason you still need to persist the dom and browser context
+between tests, you can set `testIsolation='off` on the root configuration or at
+the suite-level. For example:
+
+```js
+describe('workflow', { testIsolation: 'off' }, () => {
+  ...
+})
+```
+
+It is important to note that while turning test isolation `off` may improve the
+overall performance of end-to-end tests, previous tests could be impact the
+browser state. It is important to be extremely mindful of how test are written
+when using this mode and ensure tests continue to run independent from one
+other.
+
+<Badge type="danger">For example</Badge>the following tests are not independent
+nor deterministic:
+
+```js
+describe('workflow', { testIsolation: 'off' }, () => {
+  it('logs in', () => {
+    cy.visit('my-app.com/log-in)
+    cy.get('username').type('User1')
+    cy.get('password').type(Cypress.env('User1_password'))
+    cy.get('button#login').click()
+    cy.contains('User1')
+  })
+
+  it('clicks user profile', () => {
+    cy.get('User1').find('#profile_avatar).click()
+    cy.contains('Email Preferences')
+  })
+
+  it('updates profile', () => {
+    cy.get('button#edit')
+    cy.get('email').type('user1@email.com')
+    cy.get('button#save').click()
+  })
+})
+```
+
+In the above example, each test is relying on the previous test to be
+_successful_ to correctly execute. If at any point, the first or second test
+fails, the sequential test(s) will automatically fail and provided un-reliable
+debugging errors since the errors are representative of the previous test.
+
+The best way to ensure your tests are independent is to add a `.only()` to your
+test and verify it can run successfully without the test before it.
+
+### Alias Behaviors Changes
+
+Cypress always re-queries aliases when they are referenced. This can result in
+certain tests that use to pass failing. For example,
+
+```js
+cy.findByTestId('popover')
+  .findByRole('button', { expanded: true })
+  .as('button')
+  .click()
+
+cy.get('@button').should('have.attr', 'aria-expanded', 'false')
+```
+
+previously passed, because the initial button was collapsed when first queried,
+and then later expanded. However, in Cypress 12, this test fails because the
+alias is always re-queried from the DOM, effectively resulting in the following
+execution:
+
+```js
+cy.findByTestId('popover').findByRole('button', { expanded: true }).click()
+
+cy.findByTestId('popover')
+  .findByRole('button', { expanded: true }) // A button which matches here (is expanded)...
+  .should('have.attr', 'aria-expanded', 'false') // ...will never pass this assertion.
+```
+
+You can rewrite tests like this to be more specific; in our case, we changed the
+alias to be the first button rather than the unexpanded button.
+
+```js
+cy.findByTestId('popover').findAllByRole('button').first().as('button')
+```
+
+### Command / Cypress API Changes
```

Comment:
It feels like this should be at a different header level than the elements below it, or removed entirely (the ### headers below seem like they stand on their own just fine).